From f84135837963a3a3310afebe8abe6fdf2925a653 Mon Sep 17 00:00:00 2001
From: Ben Hilburn <ben.hilburn@ettus.com>
Date: Fri, 28 Aug 2015 21:01:44 -0700
Subject: [PATCH 11/17] Fixes Cov Defect 1046011: Resource Leak

Simple issue of a file handle not getting closed in `interleaver.cc`.
---
 gr-trellis/lib/interleaver.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gr-trellis/lib/interleaver.cc b/gr-trellis/lib/interleaver.cc
index f24d625..9653ee1 100644
--- a/gr-trellis/lib/interleaver.cc
+++ b/gr-trellis/lib/interleaver.cc
@@ -94,6 +94,8 @@ namespace gr {
       for(int i=0;i<d_K;i++) {
 	d_DEINTER[d_INTER[i]]=i;
       }
+
+      fclose(interleaverfile);
     }
 
     //######################################################################
-- 
2.1.4

